### PR TITLE
Added Feature for using configuration from toml file and parsing it.

### LINKED ISSUE #35: Feature: Support using a TOML "dotfile" config file in the user's homedir
**Is your feature request related to a problem? Please describe.**
This feature is not related to a problem.

### PR DESCRIPTION
**Pull request by hpatel292-seneca**
This fixes #35

### CODE REVIEW

**Comment by uday-rana** [comment]
Hi, could you please explain your changes?

The diff lights up the whole file.

Thanks.

**Comment by hpatel292-seneca** [comment]
## description
---
This PR fixes #36  and introduces support for TOML configuration file in the CLI tool. Users can now specify default settings in a `.codeshift.toml` file located in their home directory, which enhances usability by reducing the need for user prompts when required parameters are not provided.

## Changes
---
1) Installed `smol-toml` npm package.
2) Update `.gitignore` to ignore `.codeshift.toml` file.
3) Add `loadConfig()` function that will load config file and update `defaultConfig` variable
image
4) Add `loadConfig()` function call at start of `runApp()`.
image
5) update `-o/--output` and `-t/--token-usage` option to have a default value which we parsed from toml config.
image
6) I moved `program.parse(process.argv)` into `runApp` to ensure configuration is loaded before parsing command-line arguments.

File: src/index.js [hunk 1]
```diff
@@ -5,6 +5,11 @@ const fs = require("node:fs/promises");
 const { program } = require("commander");
 const { version, name, description } = require("../package.json");
 const OpenAI = require("openai");
+const toml = require("smol-toml");
+const path = require("path");
+const os = require("os");
+
+const configFilePath = path.join(__dirname, "../.codeshift.toml");
```
Comment (uday-rana) [inline src/index.js]: Should look in the user's home directory `~`

See these links for a good way to do it:

https://stackoverflow.com/questions/9080085/node-js-find-home-directory-in-platform-agnostic-way

https://nodejs.org/docs/latest-v20.x/api/os.html#oshomedir

**Review by uday-rana — CHANGES_REQUESTED** [review]

**Comment by hpatel292-seneca** [comment]
hi @uday-rana, I have commited changes, Please review

File: src/index.js [hunk 2]
```diff
@@ -5,6 +5,13 @@ const fs = require("node:fs/promises");
 const { program } = require("commander");
 const { version, name, description } = require("../package.json");
 const OpenAI = require("openai");
+const toml = require("smol-toml");
+const path = require("path");
+const os = require("os");
+const { Console } = require("node:console");
```
Comment (uday-rana) [inline src/index.js]: Unused import

File: src/index.js [hunk 3]
```diff
@@ -27,144 +34,178 @@ const providers = {
...
-  .option("-o, --output <filename>", "specify filename to write output to")
-  .option("-t, --token-usage", "report token usage")
-  .argument("<output-language>", "language to transform code to")
-  .argument("<input-files...>", "source files to read")
-  .action(async (outputLang, inputFiles) => {
-    if (!process.env.API_KEY) {
-      console.error(`missing expected env var: "API_KEY"`);
-      process.exit(20);
-    }
+// default values
```
Comment (uday-rana) [inline src/index.js]: - Capitalize first letter for consistency
- Make more descriptive (Default values for what?)

**Review by uday-rana — CHANGES_REQUESTED** [review]
Feature works good, just needs some small changes

File: src/index.js (hunk 3 above)
Comment (hpatel292-seneca) [reply src/index.js]: Done.

**Review by uday-rana — APPROVED** [review]
Looks good, thanks for making the changes

**Pull request merged**